### ISSUE TITLE
多路径推流，聚合路由器方案

### ISSUE THREAD

**Issue by winlinvip** [ossrs/srs#2637]
现在每周会做一场直播，主要是SRS的答疑，推RTMP流到视频号，发现有时候网络抖动会卡顿。

上行推流如果网络不好，会导致所有人都卡，因此基本上直播都需要保障上行推流，除非是一个网络摄像头24小时推流直播没人看的那种场景。

目前市面上的方案是聚合路由器，一般是在路由器上插多张网卡，部署一个专用的服务器来聚合这些网卡的流，然后推到直播源站。这里有几个问题：

1. 这是路由器厂商的方案，完全物理层的多路径传输，其实如果做直播专用的多路径，做应用层的多路径推流，会更简单也可以有更多的业务处理。
2. 多路径并没有冗余，如果遇到某个链路质量不好或抖动，需要比较复杂的算法来处理。比如一条链路的RTT从10ms变成100ms，这个链路质量怎么计算，比较复杂，因为从某些包看不到最终的结果，如果采用冗余推流，就可以只看最终这些流的FPS是否正常，选择FPS比较稳定的一路流转推就行。
3. 一般服务器是路由器厂商部署，自己部署比较麻烦，而且是链路层的服务器。不如部署专用的视频服务器比较直观，比如会有很多视频的指标。

所以我觉得应用层的聚合路由器方案，会更好一些，具体方案参考：https://www.processon.com/view/link/61513ae807912903eff891c6

**Comment by online2311** [comment]
I can't understand some of the multi-line configurations for the SRS live broadcast aggregation router.

**Comment by MattUan** [comment]
Looking at your architecture diagram, the approximate effect achieved is that the SRS server shunts the push and the receiving end selects the best link for forwarding? In this case, it serves as a redundancy, while MPTCP serves to stack the speed of multiple links, such as stacking three 10M broadband lines to achieve 30M to increase available bandwidth. I've looked at the latest documentation for openmptcprouter, and it seems to have added hot standby redundancy features. I haven't studied the specific implementation yet, but in any case, it should be a different direction from what you're saying. However, integrating SRS aggregation should effectively reduce fault points and lower latency?

**Comment by winlinvip** [comment]
@MattUan Yes, it's different from MPTCP. If it were the same, there would be no need to do it, and MPTCP would be good enough.

What I want to do mainly addresses the redundancy issue when bandwidth requirements are met, and of course, redundancy can also be achieved with multiple camera positions.

In addition, MPQUIC may be more suitable than MPTCP, but it is not yet mature.
